### ISSUE TITLE
Optional TECHNICAL_MESSAGES parameter on Validator::make to repress stripping of underscores from attribute placeholder

### ISSUE THREAD

**Issue by martinjoiner** [laravel/internals#837]
I am offering to fork, code and create pull-request to implement the following feature:

An enhancement to Validation whereby the replacement of underscores with spaces in the attribute name can be repressed. 

Currently line 240 of `/Validation/Concerns/FormatsMessages.php` is where the operation takes place and it cannot be disabled/overridden. 

**Use case example:** 

For when feedback messages are to be consumed by developers (not the general public) and the messages should be more technical and specific. Think public API routes. 

So for the message:

```
'The :attribute field is required.'
```

By default Laravel would still return:

> The vehicle name field is required.

But with the TECHNICAL_MESSAGES flag set it would return the more technical:

> The vehicle_name field is required.

**Proposed implementation**

I am proposing adding the flag as a constant on the Validator class for readability and then the flag would be provided in the `customAttribute` array as the fourth argument of `Validator::make()`. The FormatsMessages concern would then check if it is set before continuing to strip underscores. 

An implementation might look something like this: 

```
Validator::make( $request->all(), $params, [], [ Validator::TECHNICAL_MESSAGES ] );
```

Please, if you think this is a good enhancement just give me the thumbs up and I will do the work. 

Thanks.

**Comment by sisve** [comment]
This information is already present in the default HTTP 422 error result, which is the error you get when a validation error occurs and you accept json responses (the request has `Accept: application/json`).

I see no reason to implement what you describe based on the use case you provided.

    {
        "message": "The given data was invalid.",
        "errors": {
            "vehicle_name": [
                "The vehicle name field is required"
            ]
        }
    }

**Comment by martinjoiner** [comment]
@sisve Are you suggesting that as a developer implementing a call to an API, if you make a typo you would prefer the error message "The vehicle name field is required" to "The vehicle_name field is required"?  Because your solution to rely on the default still strips that underscore.

**Comment by sisve** [comment]
Yes.

The rule that failed the validation is present in the response. If you're looking at the http requests in your browser you'll also see what data you're posting. Why would you need to _also_ include it in the error message in a way that requires code changes in every api?

This change can already be achieved on an application-wide basis by writing a custom validator and overriding the getDisplayableAttribute method.

    class CustomValidator extends Validator {
        public function getDisplayableAttribute($attribute) {
            return 'EXAMPLE:' . $attribute;
        }
    }

Results in:

    {
        "message": "The given data was invalid.",
        "errors": {
            "vehicle_name": [
                "The EXAMPLE:vehicle_name field is required"
            ]
        }
    }

**Comment by martinjoiner** [comment]
I was returning a HTML page via a view file for the 412 Precondition Failed response in which I displayed the error messages as list items. Providing a JSON response as you suggest is a much better idea and in that context your argument about the keys being visible makes sense. 

I also tried your solution for repressing the underscore substitution by make TechnicalValidator that extends the Validator class and overrides the `getDisplayableAttribute()` method but I got stuck after receiving "Call to undefined method App\Services\TechnicalValidator::setPresenceVerifier()".
